packages/@aws-cdk/core/lib/private/cloudformation-lang.ts

```diff
@@ -211,12 +214,34 @@ function tokenAwareStringify(root: any, space: number, ctx: IResolveContext) {
         pushLiteral('"');
         pushIntrinsic(deepQuoteStringLiterals(intrinsic));
         pushLiteral('"');
-        break;
-
-      default:
+        return;
+
+      case ResolutionTypeHint.LIST:
+        // We need this to look like:
+        //
+        //    '{"listValue":' ++ STRINGIFY(CFN_EVAL({ Ref: MyList })) ++ '}'
+        //
+        // However, STRINGIFY would need to execute at CloudFormation deployment time, and that doesn't exist.
+        //
+        // We could *ALMOST* use:
+        //
+        //   '{"listValue":["' ++ JOIN('","', { Ref: MyList }) ++ '"]}'
+        //
+        // But that has the unfortunate side effect that if `CFN_EVAL({ Ref: MyList }) == []`, then it would
+        // evaluate to `[""]`, which is a different value. Since CloudFormation does not have arbitrary
+        // conditionals there's no way to deal with this case properly.
```

Comment:
@rix0rrr  I think I found a way with a `Fn::Join` on a `Fn::Split` acting like a replace:


```ts
cdk.Fn.join('', cdk.Fn.split('""', myString)) // acts like a replace
```


For #13465 and the issue with the array of ENIs, it looks like that

```ts
[cdk.Token.asList(cdk.Fn.join('', cdk.Fn.split('""', cdk.Stack.of(stack).toJsonString(cdk.Fn.join('","', endpoint.vpcEndpointNetworkInterfaceIds)))))]
```

Gives the following CF
```json
{
  "Fn::Join": [
    "",
    [
      "{\"action\":\"describeNetworkInterfaces\",\"service\":\"EC2\",\"parameters\":{\"NetworkInterfaceIds\":[",
      {
        "Fn::Join": [
          "",
          {
            "Fn::Split": [
              "\"\"",
              {
                "Fn::Join": [
                  "",
                  [
                    "\"",
                    {
                      "Fn::Join": [
                        "\\\",\\\"",
                        {
                          "Fn::GetAtt": [
                            "OtherEndpoint88C37F72",
                            "NetworkInterfaceIds"
                          ]
                        }
                      ]
                    },
                    "\""
                  ]
                ]
              }
            ]
          }
        ]
      },
      "]},\"physicalResourceId\":{\"id\":\"physical-id\"},\"outputPath\":\"NetworkInterfaces.0.PrivateIpAddress\"}"
    ]
  ]
}
```

Successfully tested this with a `GatewayVpcEndpoint` which has no ENIs and hence returns an empty list for the `vpcEndpointNetworkInterfaceIds` attribute.

WDYT?

Reply:
EHRMAGERD

Reply:
You are like a wizard.

A dirty wizard...

Reply:
I like it though.

Reply:
Wait no. This solution will break when presented with `[""]`.

That value would successfully make it through the transformation and then be left as `[]`.

Reply:
ahh 😄 

You're right, there's may be something to do if we first join the array with something else than `","`... I need to think about it 🤔

Reply:
Bu then maybe the CR is the right solution

Reply:
I'm also thinking about it... afraid nothing's to be done, as:

```
JOIN(<whatever>, []) == JOIN(<whatever>, [""]) == ""
```

Once we've JOINed, we've lost the ability to make the distinction between `[]` and `[""]`... right? 

(I'm starting to doubt myself now)

Reply:
I don't think there's a way to do it. You loose information after the first join:

```ts
[].join('","') === [''].join('","')
``` 

and the only "first operation" you can do on the array with CF is a `Fn::Join`.

Reply:
OK our comments crossed each other... I don't think it's possible in pure CF. This is sad 😢 .

Reply:
IF ONLY we had a functional `{ Fn::If }`...